x-pack/plugins/apm/server/lib/helpers/get_bucket_size/index.ts

```diff
@@ -13,17 +13,25 @@ export function getBucketSize({
   start,
   end,
   numBuckets = 100,
+  minBucketSize,
 }: {
   start: number;
   end: number;
   numBuckets?: number;
+  minBucketSize?: number;
 }) {
   const duration = moment.duration(end - start, 'ms');
   const bucketSize = Math.max(
     calculateAuto.near(numBuckets, duration).asSeconds(),
     1
   );
-  const intervalString = `${bucketSize}s`;
+
+  if (minBucketSize && bucketSize < minBucketSize) {
+    return {
+      bucketSize: minBucketSize,
+      intervalString: `${minBucketSize}s`,
+    };
+  }
```

Comment:
Is this block necessary? You can set the `bucketSize` in the `Math.max` statement like I suggested in another comment